retail-managment-system/rms-ui/src/app/pages/store/store/store.component.ts

```diff
@@ -0,0 +1,190 @@
+        this.translate.instant('store.close'),
+        {
+          duration: 2000,
+        }
+      ); 
+    }
+  })
+ }
+
+  storgeForm(){
+    this.storeForm = this.build.group({
+      id:[''],
+      name:['',Validators.maxLength(100)],
+      address:['',[Validators.maxLength(100),Validators.required]],
+      responsible:['']
+    })
+  }
+
+  onPaginationChange(event: PageEvent) {
+
+    this.page  = event.pageIndex;
+    this.size = event.pageSize;
+    this.getStores(this.currentResponsible)
+  }
+  
+  fetchData(store:Store): void {
+    this.isButtonVisible = false;
+    this.storeForm.patchValue(store)
+    this.currentData = store
+  }
+
+  updateStore() {
+    this.isButtonVisible = true;
+    this.submit = true;
+    this.storeRepository.update(this.storeForm.value).subscribe(
+      () => {
+        this.getStores(this.currentResponsible)
+        this.submit = false;
+        this.snackBar.open(
+          this.translate.instant('store.updated-successfuly'),
+          this.translate.instant('store.close'),
+          {
+            duration: 2000,
+          }
+        );
+      },
+      () => {
+        this.submit = false;
+      }
+    );
+  }
+
+  addStore() {
+    this.isButtonVisible = true;
+    this.submit = true;
+    this.storeRepository.add(this.storeForm.value).subscribe(
+      () => {
+        this.getStores(this.currentResponsible)
+        this.submit = false;
+        this.snackBar.open(
+          this.translate.instant('store.added-successfuly'),
+          this.translate.instant('store.close'),
+          {
+            duration: 2000,
+          }
+        );
+      },
+      () => {
+        this.submit = false;
+      }
+    );
+  }
+
+  deleteStore(store: Store) {
+    this.storeRepository.delete(store.id).subscribe(() => {
+      this.getStores(this.currentResponsible)
+      this.snackBar.open(
+        this.translate.instant('store.delete-successfuly'),
+        this.translate.instant('store.close'),
+        {
+          duration: 2000,
+        }
+      );
+    });
+  }
+
+  onSubmit() {
+    this.storeForm.markAllAsTouched();
+    if (this.storeForm.valid) {
+      this.storeForm.controls['id'].value
+        ? this.updateStore()
+        : this.addStore();
+      this.storeForm.reset();
+    }
+  }
+
+  resetForm() {
+    this.storeForm.controls['id'].value
+      ? this.fetchData(this.currentData)
+      : this.storeForm.reset();
+  }
+  compareFn(a: Employee, b: Employee) {
+    if (!a || !b) return false;
+    return a.id === b.id;
+  }
+
+  restartForm(): void {
+    this.storeForm.reset();
+  }
+
+  openConfirmationDialog(store: Store) {
+    let dialogRef = this.dialog.open(ConfirmDialogComponent);
+    dialogRef.afterClosed().subscribe((result) => {
+      if (result === 'yes') {
+        this.deleteStore(store);
+      }
+    });
+  }
+
+  getAllEmployee(): void {
+    this.employeeRepository.getList().subscribe((result) => {
+      this.employees = result.data;
```

Comment:
where is pagination?

Reply:
I think I don't need pagination as I just use getAllEmployee() to display data in select form field in the form